### PR TITLE
Update ANTLR grammar from compass tool

### PR DESCRIPTION
**Pull request by timchang514**
### Description

Taking the latest updates from babelfish_compass to update the parser and lexer; fix the corresponding areas in the parser code and error handler.

https://github.com/babelfish-for-postgresql/babelfish_compass/blob/v.2022-07/src/main/parser/TSQLParser.g4
https://github.com/babelfish-for-postgresql/babelfish_compass/blob/v.2022-07/src/main/parser/TSQLLexer.g4

Task: BABEL-3390

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -5310,7 +5310,22 @@ post_process_create_table(TSqlParser::Create_tableContext *ctx, PLtsql_stmt_exec
 		post_process_column_definition(cdctx, stmt, baseCtx);
 
 	// viist options in index specification
-	for (auto ictx : ctx->table_indices())
+	for (auto ictx : ctx->inline_index())
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:5313]: Can we use `post_process_inline_index()` here?

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -5310,7 +5310,22 @@ post_process_create_table(TSqlParser::Create_tableContext *ctx, PLtsql_stmt_exec
...
+			removeCtxStringFromQuery(stmt->sqlstmt, ictx->storage_partition_clause()[0], baseCtx);
+		}
+
+		if (ictx->clustered() && ictx->clustered()->CLUSTERED())
+			removeTokenStringFromQuery(stmt->sqlstmt, ictx->clustered()->CLUSTERED(), baseCtx);
+		if (ictx->clustered() && ictx->clustered()->NONCLUSTERED())
+			removeTokenStringFromQuery(stmt->sqlstmt, ictx->clustered()->NONCLUSTERED(), baseCtx);
+	}
+
+	for (auto ictx : ctx->table_constraint())
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:5318]: Same here for `post_process_table_constraint()`

**Review by 2jungkook — COMMENTED** [review]
In the PR description, it would be nice to add the followings
- Task: BABEL-3390
- Pointer to compass files referenced by this PR

**Review by priyansx — APPROVED** [review]
LGTM

**Review by 2jungkook — APPROVED** [review]

**Review by macwoody — APPROVED** [review]

**Pull request merged**